includes/Hooks.php

```diff
@@ -248,6 +251,54 @@ public static function onCodeEditorGetPageLanguage( $title, &$lang, $model, $for
 		return true;
 	}
 
+    /**
+     * @param $old
+     * @param $new
+     * @param $userIdentity
+     * @param int $pageid
+     * @param int $redirid
+     * @param string $reason
+     * @param $revision
+     * @return true
+     */
+    public static function onPageMoveCompleting($old, $new, $userIdentity, int $pageid, int $redirid, string $reason, $revision) {
+        $config = self::getConfig();
+        if ($revision->getContent( SlotRecord::MAIN )->getModel() === 'sanitized-css' ) {
+            return true;
+        }
+
+        $text = $revision->getContent( SlotRecord::MAIN )->getText();
+        $content = new TemplateStylesContent( $text );
+        $wikiPageFactory  = MediaWikiServices::getInstance()->getWikiPageFactory();
+        $userFactory  = MediaWikiServices::getInstance()->getUserFactory();
+        $namespace = $new->getNamespace();
+        $titleDBKey = $new->getDBkey();
+
+        if ( $namespace !== $config->get( 'TemplateStylesDefaultNamespace' ) || substr( $new->getText(), -4 ) !== '.css' ) {
```

Comment:
nitpick: we could use str_ends_with here

Reply:
another nitpick: this if could probably go before we initialize: $content, $wikiPageFactory, $titleDBKey,  $userFactory